sql/core/src/test/scala/org/apache/spark/sql/VariantSuite.scala

```diff
@@ -192,4 +192,22 @@ class VariantSuite extends QueryTest with SharedSparkSession {
       }
     }
   }
+
+  test("group/order/join variant are disabled") {
```

Comment:
Thanks for the tests!

What about sort by? or window partition by, or window order by? Some examples:

```
select parse_json('') sort by 1

with t as (select 1 as a, parse_json('') as v)
select rank() over (partition by v order by a)

with t as (select 1 as a, parse_json('') as v)
select rank() over (partition by a order by v)
```

Reply:
This is a very good point. Actually, the third statement will not fail as expected, and it is really an error in the type check for `Window`. Both `partitionSpec` and `orderSpec` must be orderable, but the current type check only includes `orderSpec`. If the `partitionSpec` contains a map type, the query will fail later in an optimizer rule with `INTERNAL_ERROR`. I will create another PR to fix the type check for `Window`.

Reply:
If we can merge https://github.com/apache/spark/pull/45730 before this PR, we can add the second statement to the tests.